app/test/features/calendar_sync/calendar_sync_permission_page_test.dart

```diff
@@ -0,0 +1,24 @@
+import 'package:acter/common/toolkit/buttons/primary_action_button.dart';
+import 'package:acter/features/calendar_sync/calendar_sync_permission_page.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import '../../helpers/test_util.dart';
+
+void main() {
+  testWidgets('renders correctly with all UI elements', (tester) async {
```

Comment:
nice to have UI tests but I think it's not enough to fully confide that functionality is working...